### PR TITLE
Authorize comment `#create` api endpoint

### PR DESCRIPTION
**Pull request by krauselukas**
Currently there is not authorization going on, in some cases the `CommentLockingValidator` still prevents comments from being created if permission is insufficient in certain cases. But some things are not handled, for example when a user is blocked for commenting.

This should be handled on the controller level using our pundit policy.

@a-chris thanks bringing this up

### CODE REVIEW

File: src/api/app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -8,7 +8,9 @@ def index
   end
 
   def create
-    @obj.comments.create!(body: request.raw_post, user: User.session!, parent_id: params[:parent_id])
+    comment = @obj.comments.new(body: request.raw_post, user: User.session!, parent_id: params[:parent_id])
+    authorize comment, :create?
```
Comment (danidoni) [inline src/api/app/controllers/comments_controller.rb:12]: I believe we should test the policy `:create` method

Comment (krauselukas) [reply src/api/app/controllers/comments_controller.rb:12]: I haven't checked if this policy is tested... just wanted to fix the endpoint. I will come up with a little test for this

Comment (danidoni) [reply src/api/app/controllers/comments_controller.rb:12]: It is not. Maybe in another PR that this will depend on?

Comment (krauselukas) [reply src/api/app/controllers/comments_controller.rb:12]: I create the specs in this PR @danidoni

**Review by danidoni — APPROVED** [review]

**Pull request merged**